Sources/HTTP/HTTPStreamingParser.swift

```diff
@@ -284,21 +304,29 @@ public class StreamingParser: HTTPResponseWriter {
     func bodyReceived(data: UnsafePointer<Int8>?, length: Int) -> Int32 {
         processCurrentCallback(.bodyReceived)
         guard let data = data else { return 0 }
+        if shouldStopProcessingBody {
```

Comment:
Optimally, this should be before the assignment.

Reply:
Locking the semaphore to check the Bool value should take precedence over the nil-check?

Reply:
Sorry, I'm getting ahead of myself. I'm not convinced we need any lock protection around a stop flag. We're dealing with two serial queues - one reader and one writer. The only logic that runs on the writer queue is the writeXXXX() calls, so if reads and updates to the stop flag are only by code executing on the reader queue then we don't need a lock.

If we don't need to lock it, it should optimally be before the nil check and assign.